### PR TITLE
MNT: Remove uproot-base build ouptut

### PR DESCRIPTION
**Pull request by matthewfeickert**
* As of `v5.4.0` the requirements of the `uproot-base` and `uproot` build outputs are identical. This removes the need for `uproot-base` and the feedstock can now produce just the `uproot` build output.
   - c.f. PR https://github.com/conda-forge/uproot-feedstock/pull/157
* Rerender the `meta.yaml` using `grayskull pypi uproot`.
* Use the stable docs URL.
* Bump the build number.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml
Comment (matthewfeickert) [inline recipe/meta.yaml:13]: I originally had

```yaml
  skip: true # [py<38]
```

here but the linter failed with

> Non noarch packages should have python requirement without any version constraints.

and the CI was failing for macOS (for unknown reasons).

When I changed this back to

```yaml
  noarch: python
```

the linter is failing with 

> `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.

While conda-forge builds only for Python 3.9+ now, it seems like you should still want to ensure that the lower bound constraints are enforced to avoid people getting installations on the wrong CPython.

@conda-forge/help-python @chrisburr can you offer advice here?

Comment (matthewfeickert) [reply recipe/meta.yaml:13]: From https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python I'm confused about the defintion of "selectors" as it mentions:

> In order to qualify as a noarch python package, all of the following criteria must be fulfilled:
> 
> * No compiled extensions
> * No post-link or pre-link or pre-unlink scripts
> * No OS-specific build scripts
> * No python version specific requirements
> * No skips except for python version. If the recipe is py3 only, remove skip statement and add version constraint on python in `host` and `run` section.
> * `2to3` is not used
> * `scripts` argument in `setup.py` is not used
> * If `console_scripts` `entry_points` are defined in `setup.py` or `setup.cfg`, they are also [listed](https://conda.io/projects/conda-build/en/stable/resources/define-metadata.html#python-entry-points) in the `build` section of `meta.yaml`
> * No activate scripts

but then also mentions

> [!NOTE] 
> While `noarch: python` does not work with selectors, it does work with version constraints. `skip: True # [py2k]` can be replaced with a constrained python version in the `host` and `run` subsections: say `python >=3` instead of just `python`.

This seems to be exactly what we're doing with

```yaml
requirements:
  host:
    - python >=3.8
    - hatchling >1.10.0
    - hatch-vcs
    - pip
  run:
    - python >=3.8
```

so I'm confused why the linter is rejecting this.

Comment (matthewfeickert) [reply recipe/meta.yaml:13]: Oh. From the conda-forge docs FAQ [Why does conda-build ignore the py37 selector in meta.yaml?](https://conda-forge.org/docs/maintainer/maintainer_faq/#why-does-conda-build-ignore-the-py37-selector-in-metayaml) (which links to Selectors in conda-build documentation ([Preprocessing selectors](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#preprocessing-selectors))) I now realize that "selectors" are the

```
# [<selector>]
```

format.

So the selector the linter is complaining about is the `# [py<311]` in

```yaml
  run:
    - python >=3.8
    - awkward >=2.4.6
    - cramjam >=2.5.0
    - python-xxhash
    - numpy
    - fsspec
    - packaging
    - typing_extensions >=4.1.0  # [py<311]
```

So I guess that needs to be removed even and `typing_extensions` will get installed for everything?

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

**Comment by matthewfeickert** [comment]
@conda-forge/scikit-hep this is ready for review now.

File: recipe/meta.yaml
Comment (jpivarski) [inline recipe/meta.yaml:9]: I understand that there's only one package now, but why should `uproot` be hard-coded?

**Review by jpivarski — APPROVED** [review]
It looks good to me, and I'm in favor of replacing the `uproot-base` + `uproot` combination with just `uproot`.

File: recipe/meta.yaml
Comment (matthewfeickert) [reply recipe/meta.yaml:9]: @jpivarski This is the behavior of `grayskull` now. I wondered the same thing, but I haven't bothered to ask on https://github.com/conda/grayskull.

Comment (matthewfeickert) [reply recipe/meta.yaml:9]: @jpivarski c.f. https://github.com/conda/grayskull/discussions/569

Comment (matthewfeickert) [reply recipe/meta.yaml:9]: I'm going to merge this in, but if there's some answer from the `grayskull` team that makes us think we should change from what they do then we can open a fix PR.

**Pull request merged**
